android/src/main/java/com/swmansion/rnscreens/ScreenWindowTraits.kt

```diff
@@ -115,12 +116,33 @@ object ScreenWindowTraits {
                     if (translucent) {
                         ViewCompat.setOnApplyWindowInsetsListener(decorView) { v, insets ->
                             val defaultInsets = ViewCompat.onApplyWindowInsets(v, insets)
-                            defaultInsets.replaceSystemWindowInsets(
-                                defaultInsets.systemWindowInsetLeft,
-                                0,
-                                defaultInsets.systemWindowInsetRight,
-                                defaultInsets.systemWindowInsetBottom
-                            )
+
+                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                                val windowInsets =
+                                    defaultInsets.getInsets(WindowInsetsCompat.Type.statusBars())
+
+                                WindowInsetsCompat
+                                    .Builder()
+                                    .setInsets(
+                                        WindowInsetsCompat.Type.statusBars(),
+                                        Insets.of(
+                                            windowInsets.left,
+                                            0,
+                                            windowInsets.right,
+                                            windowInsets.bottom
+                                        )
+                                    )
+                                    .build()
+
+                                insets
```

Comment:
Hello @tboba 👋 

Thank you for your review! Actually I left `insets` intentionally - without this line it breaks my app (content is moved below navigationBar - I tested it on Pixel 7 Pro (Android 13), and I think @hirbod experienced the same problem without this line as well):

|Just opened a screen|Closed warning|Opened keyboard|
|---------------------|----------------|------------------|
|![telegram-cloud-photo-size-2-5397866210319782202-y](https://github.com/software-mansion/react-native-screens/assets/22820318/f3783f59-bd3d-483d-9f97-3ae8060b7c8f)|![telegram-cloud-photo-size-2-5397866210319782203-y](https://github.com/software-mansion/react-native-screens/assets/22820318/87902bfd-1469-4064-8cd0-dbe3bc2ddf03)|![telegram-cloud-photo-size-2-5397866210319782204-y](https://github.com/software-mansion/react-native-screens/assets/22820318/11ab10e1-2b12-4a3b-aaaf-cc08e1e65d2c)|

The videos that you've attached - did you test them in example from https://github.com/software-mansion/react-native-screens/pull/1802?

Reply:
Nope, I've tested it in example from #1844 - it's a similar one, but only with use of `useHeaderHeight` and one "Post". You can find original test in #1072 (as #1802 and #1844 are copies, geared intentionally for testing header height). 

That's strange 🤔 could you include your code example in TestsExample and FabricTestsExample, named as `Test1868.tsx`, so I could dive deeper into that?

Reply:
@tboba sure! Do you want me to add RNKC as dependency so you will have identical test that I'm using, right?

Reply:
@kirillzyusko actually, if this test requires to have RNKC as a dependency, let's not add it (we want to keep the dependency list as tight as possible 😶). Also, I can see you've already added that test [here](https://github.com/kirillzyusko/react-native-keyboard-controller/pull/50), right? If it's the same test, I could test this behavior there.

Reply:
> Also, I can see you've already added that test (or an example in your case) https://github.com/kirillzyusko/react-native-keyboard-controller/pull/50, right? If it's the same test, I could test this behavior there.

Yes @tboba, you can use this test 🙂 Then no further action for now is required from my side, right?

Reply:
Yup, let me investigate that 🤞

Reply:
@kirillzyusko so, I've tried to test your changes and it looks like there is no difference between statusBarTranslucent set to true from false in the current state 😕 
It looks like the line with call to `insets` is indeed setting the insets inappropriately (as it is ignoring a call to `WindowInsetsCompat`. Why is it important in your case? It seems that removing it and setting `statusBarTranslucent` to false gives expected behavior 🤔 

<img width="460" alt="image" src="https://github.com/software-mansion/react-native-screens/assets/23281839/41b038f6-b570-4502-b3c5-179b6fcca49e">

Reply:
Anyway @tboba I think you are right - the line with `insets` causes a regression and I'll delete it tomorrow (seems like RNKC also has a different problem, but I will address it separately).

Also I will try to see whether `replaceWindowInsets` can be replaced by compat API (I have an assumption that it may work fine) - will check tomorrow!

Reply:
@tboba I reverted this line. Feel free to review PR again 🙌